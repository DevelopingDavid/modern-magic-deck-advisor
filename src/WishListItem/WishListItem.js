import React, { Component } from "react";
import '../styles/main.scss';

class WishListItem extends Component {
  removeWishListItem = () => {
    const [...newWishList] = this.props.wishList;
    newWishList.splice(this.props.cardIndex, 1);
    this.props.saveArray('wishList', newWishList);
  }

  increaseWishListCount = () => {
    if (this.props.wishListCount < 4) {
      this.updateWishList(this.props.wishListCount + 1);
    }
  }

  decreaseWishListCount = () => {
    if (this.props.wishListCount > 1) {
      this.updateWishList(this.props.wishListCount - 1);
    }
  }

  updateWishList = (newCount) => {
    const newWishList = this.props.wishList.map(card => {
      if (this.props.cardName === card.cardName) {
        card.wishListCount = newCount;
      }
      return card;
    });
    this.props.saveArray('wishList', newWishList);
  }

  render() {
    return(
      <li id={this.props.cardIndex} className="wishlist--item">
        {this.props.cardName}
        <i className="fas fa-minus" onClick={this.decreaseWishListCount}/>
        <span>{this.props.wishListCount}</span>
        <i className="fas fa-plus" onClick={this.increaseWishListCount}/>
        <i className="far fa-trash-alt" onClick={this.removeWishListItem}/>
      </li>
    )
  }
}

export default WishListItem;