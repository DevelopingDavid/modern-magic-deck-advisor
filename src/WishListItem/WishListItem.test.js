import React from 'react';
import WishListItem from './WishListItem';
import { shallow } from 'enzyme';

const cardName = "Blood Moon"
const index = 0;
const wishListCount = 2;
const wishList = [{ cardName: 'Blood Moon', wishListCount: 2 }];
const saveArrayMock = jest.fn();

describe("WishListItem", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <WishListItem
        cardName={cardName}
        cardIndex={index}
        wishListCount={wishListCount}
        key={cardName}
        wishList={wishList}
        saveArray={saveArrayMock}/>
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call removeWishListItem when the trash-can is clicked', () => {
    wrapper.find('.fa-trash-alt').simulate('click');
    expect(saveArrayMock).toBeCalled();
  });

  it(`should call updateWishList with the incremented count
  when the plus sign is clicked`, () => {
  wrapper.instance().updateWishList = jest.fn();
  wrapper.update();
  wrapper.find('.fa-plus').simulate('click');
  expect(wrapper.instance().updateWishList).toBeCalledWith(3);
  });

  it(`should call updateWishList with the decremented count
    when the plus sign is clicked`, () => {
    wrapper.instance().updateWishList = jest.fn();
    wrapper.update();
    wrapper.find('.fa-minus').simulate('click');
    expect(wrapper.instance().updateWishList).toBeCalledWith(1);
  });

  it('should call saveArray when updateWishList is called', () => {
    wrapper.instance().updateWishList(3);
    expect(saveArrayMock).toBeCalled();
  });
});