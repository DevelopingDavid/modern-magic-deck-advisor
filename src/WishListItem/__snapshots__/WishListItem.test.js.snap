// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WishListItem should match the snapshot 1`] = `
<li
  id={0}
>
  Blood Moon
  <i
    className="far fa-trash-alt"
    onClick={[Function]}
  />
</li>
`;
